ArduSub/Parameters.cpp

```diff
@@ -896,3 +896,152 @@ void Sub::convert_old_parameters()
+            JSButton::button_function_t::k_servo_3_min_momentary,
+            JSButton::button_function_t::k_servo_3_max_momentary,
+            JSButton::button_function_t::k_servo_3_min_toggle,
+            JSButton::button_function_t::k_servo_3_max_toggle
+        }
+    };
+
+    // Target actuator functions for assignment
+    static constexpr SRV_Channel::Function actuator_functions[NUM_LEGACY_ACTUATORS] = {
+        SRV_Channel::Function::k_actuator1,
+        SRV_Channel::Function::k_actuator2,
+        SRV_Channel::Function::k_actuator3
+    };
+
+    // Process legacy channels 9-11
+    for (uint8_t actuator_idx = 0; actuator_idx < NUM_LEGACY_ACTUATORS; actuator_idx++) {
+        const uint8_t channel = FIRST_LEGACY_CHANNEL + actuator_idx;
+        const auto target_function = actuator_functions[actuator_idx];
+
+        // Skip if actuator function already assigned to any channel
+        uint8_t existing_channel;
+        if (SRV_Channels::find_channel(target_function, existing_channel)) {
+            continue;
+        }
+
+        // Skip if channel is not disabled
+        if (SRV_Channels::channel_function(channel - 1) != SRV_Channel::Function::k_none) {
+            continue;
+        }
+
+        // Check if any servo/actuator buttons are assigned
+        bool has_assigned_button = false;
+        for (size_t func_idx = 0; func_idx < FUNCTIONS_PER_SERVO; func_idx++) {
+            if (sub.jsbutton_function_is_assigned(servo_functions[actuator_idx][func_idx])) {
+                has_assigned_button = true;
+                break;
+            }
+        }
+
+        if (has_assigned_button) {
+            // Assign actuator function to preserve legacy behavior
+            char param_name[20];
+            snprintf(param_name, sizeof(param_name), "SERVO%u_FUNCTION", channel);
+            AP_Param::set_and_save_by_name(param_name, static_cast<int>(target_function));
+        }
+    }
+}
+
+void Sub::update_lights_from_rcin()
+{
+    /*
+    Maps older systems from using RCIN9 and RCIN10 to using lights1 and lights2.
+    This is only done if there are joystick buttons assigned to the lights functions and there are no channels assigned to the lights functions.
+    */
+
+    const uint8_t NUM_LIGHTS = 2;
+    const uint8_t FUNCTIONS_PER_LIGHT = 3;
+    static constexpr JSButton::button_function_t lights_button_functions[NUM_LIGHTS][FUNCTIONS_PER_LIGHT] = {
+        {
+            JSButton::button_function_t::k_lights1_brighter,
+            JSButton::button_function_t::k_lights1_dimmer,
+            JSButton::button_function_t::k_lights1_cycle,
+        },
+        {
+            JSButton::button_function_t::k_lights2_brighter,
+            JSButton::button_function_t::k_lights2_dimmer,
+            JSButton::button_function_t::k_lights2_cycle,
+        }
+    };
+
+    const SRV_Channel::Function lights_functions[NUM_LIGHTS] = {
+        SRV_Channel::Function::k_lights1,
+        SRV_Channel::Function::k_lights2
+    };
+
```

Comment:
```suggestion
// Legacy, overloaded and hardcoded RCIN passthrough outputs
```
For the logical flow of old to new, I think it makes sense to define/specify the old functions before the new dedicated ones.